app/src/test/java/com/example/bottomnav/DataTest.java

```diff
@@ -0,0 +1,24 @@
+package com.example.bottomnav;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class DataTest {
+    @Test
+    public void randomData() {
+        String data = "t056111";
+        CAN_Data newData = CAN_Data.decode(data);
+        System.out.println(newData.getData());
+        assertEquals( 0x11 , newData.getData()[0]);
+        assertEquals( 86 , newData.getId());
+    }
+    @Test
+    public void BIGData() {
+        String data = "t0568111111111111111A";
+        CAN_Data newData = CAN_Data.decode(data);
+        System.out.println(newData.getData());
+        assertEquals( 0x11, newData.getData()[0]);
+        assertEquals( 8, newData.getData().length);
+    }
+}
```

Comment:
To be thorough, can you also add tests for the extended case?

In general, I'd advocate unit testing all code paths that are easily testable. I believe that in software engineering practice it isn't uncommon to have as much (or even more) testing code as actual application code.

Also, consider testing malformed inputs. Because of the nature of a wireless connection, it may be possible to get partial (possibly even corrupted) data, so your parser needs to be tolerant. It doesn't need to decode the data successfully (which would be impossible if it were corrupted), but it does need to not crash and not give misleading results.